File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/drivers/default_driver_impl/daps/DefaultDapsDriverConfig.java
package de.fhg.aisec.ids.idscp2.drivers.default_driver_impl.daps;

import org.checkerframework.checker.nullness.qual.NonNull;

import java.nio.file.Path;
import java.nio.file.Paths;

/**
 * A Configuration class for the DefaultDapsDriver
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
public class DefaultDapsDriverConfig {

    @NonNull
    private String dapsUrl = "https://daps.aisec.fraunhofer.de";
    @NonNull
    private Path keyStorePath = Paths.get("DUMMY-FILENAME.p12");
    @NonNull
    private char[] keyStorePassword = "password".toCharArray();
    @NonNull
    private String keyAlias = "1";
    @NonNull
    private char[] keyPassword = "password".toCharArray();
    @NonNull
    private Path trustStorePath = Paths.get("DUMMY-FILENAME.p12");
    @NonNull
    private char[] trustStorePassword = "password".toCharArray();

    public static class Builder {
        @NonNull
        private final DefaultDapsDriverConfig config = new DefaultDapsDriverConfig();

        @NonNull
        public Builder setDapsUrl(String dapsUrl) {
            this.config.dapsUrl = dapsUrl;
            return this;
        }

        @NonNull
        public Builder setKeyStorePath(Path path) {
            this.config.keyStorePath = path;
            return this;
        }

        @NonNull
        public Builder setKeyStorePassword(char[] password) {
            this.config.keyStorePassword = password;
            return this;
        }

        @NonNull
        public Builder setKeyAlias(String alias) {
            this.config.keyAlias = alias;
            return this;
        }

        @NonNull
        public Builder setKeyPassword(char[] password) {
            this.config.keyPassword = password;
            return this;
        }

        @NonNull
        public Builder setTrustStorePath(Path path) {
            this.config.trustStorePath = path;
            return this;
        }

        @NonNull
        public Builder setTrustStorePassword(char[] password) {
            this.config.trustStorePassword = password;
            return this;
        }

        @NonNull
        public DefaultDapsDriverConfig build() {
            return config;
        }
    }

    @NonNull
    public String getDapsUrl() {
        return dapsUrl;
    }

    @NonNull
    public Path getKeyStorePath() {
        return keyStorePath;
    }

    @NonNull
    public char[] getKeyStorePassword() {
        return keyStorePassword;
    }

    @NonNull
    public String getKeyAlias() {
        return keyAlias;
    }

    @NonNull
    public char[] getKeyPassword() {
        return keyPassword;
    }

    @NonNull
    public Path getTrustStorePath() {
        return trustStorePath;
    }

    @NonNull
    public char[] getTrustStorePassword() {
        return trustStorePassword;
    }

}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/default_drivers/daps/DefaultDapsDriverConfig.kt
package de.fhg.aisec.ids.idscp2.default_drivers.daps

import java.nio.file.Path
import java.nio.file.Paths

/**
 * A Configuration class for the DefaultDapsDriver
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
class DefaultDapsDriverConfig {
    var dapsUrl = "https://daps.aisec.fraunhofer.de"
        private set
    var keyStorePath: Path = Paths.get("DUMMY-FILENAME.p12")
        private set
    var keyStorePassword: CharArray = "password".toCharArray()
        private set
    var keyAlias = "1"
        private set
    var keyPassword: CharArray = "password".toCharArray()
        private set
    var trustStorePath: Path = Paths.get("DUMMY-FILENAME.p12")
        private set
    var trustStorePassword: CharArray = "password".toCharArray()
        private set

    class Builder {
        private val config = DefaultDapsDriverConfig()
        fun setDapsUrl(dapsUrl: String): Builder {
            config.dapsUrl = dapsUrl
            return this
        }

        fun setKeyStorePath(path: Path): Builder {
            config.keyStorePath = path
            return this
        }

        fun setKeyStorePassword(password: CharArray): Builder {
            config.keyStorePassword = password
            return this
        }

        fun setKeyAlias(alias: String): Builder {
            config.keyAlias = alias
            return this
        }

        fun setKeyPassword(password: CharArray): Builder {
            config.keyPassword = password
            return this
        }

        fun setTrustStorePath(path: Path): Builder {
            config.trustStorePath = path
            return this
        }

        fun setTrustStorePassword(password: CharArray): Builder {
            config.trustStorePassword = password
            return this
        }

        fun build(): DefaultDapsDriverConfig {
            return config
        }
    }
}